### PR TITLE
More 5.7mm guns

### PR DESCRIPTION
**Pull request by DoctorBoomstick**
#### Summary
Content "Adds a few more 5.7x28mm guns, plus their magazines and spawn locations, plus non-branded variants."

#### Purpose of change
I felt like the 5.7mm category of guns was somewhat undernourished, what with only having two guns in the game which uses this cartridge. Pluss, these weapons aren’t exactly guaranteed to pop up in a given play through most of the time given they’re some what rare spawns, which meant that unless a player happened across one of these two, than any 5.7 ammo would be relatively useless.

#### Describe the solution
I added three new weapons that use the 5.7mm cartridge and which are commonly available for Sevillian acquisition, with said weapons being the FN PS90, Kel-Tec P50, and Ruger-57. I also added non-branded variants to all of them with exception to the Ruger-57, and also added a variant for the FN P90 plus it’s magazines. I also added them to common pistol and rifle spawn locations, plus also redoing the description on the FN P90 cause I thought it’d be fun to give a little bit more detail.

#### Describe alternatives you've considered
Waving my legally distinct Harry Potter magic wand, opening a portal, yanking out a flash drive with JSON files including all the guns in the world, and creating a pull request with that.

#### Testing
Spawned in all the modified and effected guns, shot them with all the fire modes and aim settings, loaded them with loaded and unloaded mags, teleported to gun stores to observe if they spawned.

#### Additional context
https://lonestarstatefirearms.com/the-truth-about-the-ruger-57-1000-round-review/
https://www.americanrifleman.org/content/review-kel-tec-s-radical-p50/
https://fnamerica.com/products/rifles/fn-ps90-standard/

### CODE REVIEW

File: data/json/items/gun/57.json [hunk 1]
```diff
@@ -44,8 +44,17 @@
     "looks_like": "hk_mp5",
     "type": "GUN",
     "reload_noise_volume": 10,
-    "name": { "str": "FN P90" },
-    "description": "The first in a new genre of guns, termed \"personal defense weapons.\"  FN designed the P90 to use their proprietary 5.7x28mm ammunition.  It is made for firing bursts manageably.",
+    "name": { "str": "FN 5.7x28mm PDW" },
+    "description": "An FN manufactured personal defence weapon, this automatic firearm is chambered for the high-velocity proprietary 5.7x28mm cartridge.  Excepting top-mounted 50-round custom magazines which are fitted to the receiver of the weapon behind the grip, thus making it a bullpup design, this little submachine gun packs quite the nasty payload.  It’s comparatively small high-speed calibre makes it excellent at both defeating armoured opponents, and firing short bursts manageably.",
```
Comment (chaohedgehogs) [inline data/json/items/gun/57.json:48]:
```suggestion
    "description": "An FN manufactured personal defence weapon, this automatic firearm is chambered for the high-velocity proprietary 5.7x28mm cartridge.  Accepting top-mounted 50-round custom magazines which are fitted to the receiver of the weapon behind the grip, thus making it a bullpup design, this little submachine gun packs quite the nasty payload.  It’s comparatively small high-speed calibre makes it excellent at both defeating armoured opponents, and firing short bursts manageably.",
```

File: data/json/items/gun/57.json [hunk 2]
```diff
@@ -78,5 +87,135 @@
...
+    "copy-from": "fn_p90",
+    "type": "GUN",
+    "name": { "str": "Civilian FN 5.7x28mm carbine" },
+    "description": "The civilian version of FN’s older 5.7x28mm PDW, this carbine features a 16 inch long barrel, feeds from the same 50-round magazines as its predecessor, and is only capable of semi-automatic fire in order to keep it in line with American firearm restrictions.",
+    "variant_type": "gun",
+    "variants": [
+      {
+        "id": "fn_ps90",
+        "name": { "str": "FN PS90" },
+        "description": "Originally introduced in 2005, the PS90 is the civilian clone of the venerable FN P90, sharing much of the same characteristics and features as it’s fully-automatic older brother.  Chambering the same 5.7x28mm cartridge as its predecessor, in order to have made it a legally distributable carbine in the eyes of the law the PS90 was manufactured to be only capable of semi-automatic fire, with a fitted on 16 inch long barrel to sidestep NFA regulations on short Barrelled rifles.",
```
Comment (chaohedgehogs) [inline data/json/items/gun/57.json:194]:
```suggestion
        "description": "Originally introduced in 2005, the PS90 is the civilian clone of the venerable FN P90, sharing much of the same characteristics and features as it’s fully-automatic older brother.  Chambering the same 5.7x28mm cartridge as its predecessor, in order to have made it a legally distributable carbine in the eyes of the law the PS90 was manufactured to be only capable of semi-automatic fire, with a fitted-on 16-inch long barrel to sidestep NFA regulations on short-barrelled rifles.",
```

File: data/json/items/gun/57.json (hunk 1 above)
Comment (TheShadowFerret) [reply data/json/items/gun/57.json:48]: Additoinal typo fix + game set in America has American spellings

```suggestion
    "description": "An FN manufactured personal defence weapon, this automatic firearm is chambered for the high-velocity proprietary 5.7x28mm cartridge.  Accepting top-mounted 50-round custom magazines which are fitted to the receiver of the weapon behind the grip, thus making it a bullpup design, this little submachine gun packs quite the nasty payload.  Its comparatively small high-speed caliber makes it excellent at both defeating armored opponents, and firing short bursts manageably.",
```

File: data/json/items/gun/57.json [hunk 3]
```diff
@@ -78,5 +87,135 @@
...
     "faults": [ "fault_gun_blackpowder", "fault_gun_dirt", "fault_gun_chamber_spent" ],
     "pocket_data": [ { "magazine_well": "500 ml", "pocket_type": "MAGAZINE_WELL", "item_restriction": [ "fnp90mag" ] } ]
+  },
+  {
+    "id": "p50",
+    "looks_like": "glock_17",
+    "type": "GUN",
+    "reload_noise_volume": 10,
+    "name": { "str": "Kel-Tec 5.7x28mm pistol" },
+    "description": "A large retro-looking semi-automatic pistol chambered for the high-velocity 5.7x28mm cartridge, mainly constructed out of polymer based materials.  Making use of standard P90 50-round box magazines which are inserted horizontally over the grip, the top of the gun swings forwards and away from the frame to facilitate reloading.  If you were looking to add a touch of retro futuristic spice to your new life in the ruins of New England, this firearm just might become your best friend.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/57.json:97]:
```suggestion
    "description": "A large retro-looking semi-automatic pistol chambered for the high-velocity 5.7x28mm cartridge, mainly constructed out of polymer-based materials.  Making use of standard P90 50-round box magazines which are inserted horizontally over the grip, the top of the gun swings forwards and away from the frame to facilitate reloading.  If you were looking to add a touch of retro-futuristic spice to your new life in the ruins of New England, this firearm just might become your best friend.",
```

File: data/json/items/gun/57.json [hunk 4]
```diff
@@ -78,5 +87,135 @@
...
+    "variant_type": "gun",
+    "variants": [
+      {
+        "id": "p50",
+        "name": { "str": "Kel-Tec P50" },
+        "description": "Unique, adaptable, and extremely retro, the Kel-Tec P50 is a truly bizarre semi-automatic pistol.  Having been regarded as possibly one of the most innovative firearms from Kel-Tec’s entire line-up of guns, the P50 makes use of the high velocity 5.7x28mm cartridge, and feeds from standard FN P90 magazines.  Boasting ambidextrous controls, an AR-15-like T-shaped charging handle, and a reload which involves swinging the receiver and barrel assemblies up and away from the frame to insert the magazine horizontally over the grip, this weapon lends a decidedly retro futuristic touch to almost any situation.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/57.json:103]:
```suggestion
        "description": "Unique, adaptable, and extremely retro, the Kel-Tec P50 is a truly bizarre semi-automatic pistol.  Having been regarded as possibly one of the most innovative firearms from Kel-Tec’s entire line-up of guns, the P50 makes use of the high velocity 5.7x28mm cartridge, and feeds from standard FN P90 magazines.  Boasting ambidextrous controls, an AR-15-like T-shaped charging handle, and a reload which involves swinging the receiver and barrel assemblies up and away from the frame to insert the magazine horizontally over the grip, this weapon lends a decidedly retro-futuristic touch to almost any situation.",
```

File: data/json/items/gun/57.json [hunk 5]
```diff
@@ -78,5 +87,135 @@
...
+    "faults": [ "fault_gun_blackpowder", "fault_gun_dirt", "fault_gun_chamber_spent" ],
+    "pocket_data": [ { "magazine_well": "500 ml", "pocket_type": "MAGAZINE_WELL", "item_restriction": [ "fnp90mag" ] } ]
+  },
+  {
+    "id": "ruger_57",
+    "looks_like": "glock_17",
+    "type": "GUN",
+    "reload_noise_volume": 10,
+    "name": { "str": "Ruger-57" },
+    "description": "a full-sized polymer framed handgun chambered in the unusual 5.7x28mm cartridge, the Ruger-57 was one of the first commercially available 5.7mm pistols on the market, with it’s main competitor being FN’s decades old, and far pricier, Five-seveN pistol.  Featuring a lightweight ergonomic design, 1911-style thumb-safety, a high visibility green front-sight with adjustable black rear iron-sights, and a price point which hovered a good 500 dollars below its competition, the Ruger-57 made the comparatively niche 5.7x28mm round far more accessible to a wider margin of potential shooters.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/57.json:144]:
```suggestion
    "description": "A full-sized polymer framed handgun chambered in the unusual 5.7x28mm cartridge, the Ruger-57 was one of the first commercially available 5.7mm pistols on the market, with its main competitor being FN’s decades old, and far pricier, Five-seveN pistol.  Featuring a lightweight ergonomic design, 1911-style thumb-safety, a high visibility green front-sight with adjustable black rear iron-sights, and a price point which hovered a good 500 dollars below its competition, the Ruger-57 made the comparatively niche 5.7x28mm round far more accessible to a wider margin of potential shooters.",
```

> the Ruger-57 was **_one of the first_** commercially available 5.7mm pistols on the market, with its main competitor being FN’s **_decades old_**, and far pricier, Five-seveN pistol.

🤔

File: data/json/items/gun/57.json [hunk 6]
```diff
@@ -78,5 +87,135 @@
...
+        "item_restriction": [ "ruger_57mag", "ruger_57_10_round_mag" ]
+      }
+    ]
+  },
+  {
+    "id": "fn_ps90",
+    "copy-from": "fn_p90",
+    "type": "GUN",
+    "name": { "str": "Civilian FN 5.7x28mm carbine" },
+    "description": "The civilian version of FN’s older 5.7x28mm PDW, this carbine features a 16 inch long barrel, feeds from the same 50-round magazines as its predecessor, and is only capable of semi-automatic fire in order to keep it in line with American firearm restrictions.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/57.json:188]:
```suggestion
    "name": { "str": "civilian FN 5.7x28mm carbine" },
    "description": "The civilian version of FN’s older 5.7x28mm PDW, this carbine features a 16 inch long barrel, feeds from the same 50-round magazines as its predecessor, and is only capable of semi-automatic fire in order to keep it in line with American firearm restrictions.",
```

File: data/json/items/gun/57.json (hunk 2 above)
Comment (TheShadowFerret) [reply data/json/items/gun/57.json:194]:
```suggestion
        "description": "Originally introduced in 2005, the PS90 is the civilian clone of the venerable FN P90, sharing much of the same characteristics and features as it’s fully-automatic older brother.  Chambering the same 5.7x28mm cartridge as its predecessor, in order to have made it a legally distributable carbine in the eyes of the law.  The PS90 was manufactured to be only capable of semi-automatic fire, with a fitted on 16 inch long barrel to sidestep NFA regulations on short Barrelled rifles.",
```

**Comment by dseguin** [comment]
You'll need to add the new id's to `tools/json_tools/generic_guns_validator.py` to get the tests to pass.

These id's: `fn_ps90`, `p50`, `ruger_57`

Should be added to this list:

https://github.com/CleverRaven/Cataclysm-DDA/blob/a820c3e73da5e6e911d8711b57d961bf0c580a54/tools/json_tools/generic_guns_validator.py#L43-L60

**Review by dseguin — APPROVED** [review]

**Pull request merged**